packages/core/src/VDocument.ts

```diff
@@ -155,4 +153,198 @@ export class VDocument {
+            // "unlist" them.
+            this._unlist(selectedNodes);
+        } else {
+            this._convertToList(selectedNodes, type);
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return the first common ancestor of two nodes.
+     *
+     * @param a
+     * @param b
+     */
+    _commonAncestor(a: VNode, b: VNode): VNode {
+        const firstAncestors = a.ancestors();
+        const lastAncestors = b.ancestors();
+        return firstAncestors.find(ancestor => lastAncestors.includes(ancestor));
+    }
+    /**
+     * Convert the given nodes to a list with the given `htmlTag`.
+     *
+     * @param nodes
+     * @param type
+     */
+    _convertToList(nodes: Array<VNode>, type: ListType): void {
+        let newList = new ListNode(type);
+        let duplicatedNodes = [];
+
+        // Lists cannot have atomic nodes as direct children.
+        nodes = Array.from(new Set(nodes.map(node => (node.atomic ? node.parent : node))));
+
+        // If the last node is within a list, we need to split that list.
+        const last = nodes[nodes.length - 1];
+        const lastListAncestor = this._listAncestor(last);
+        if (lastListAncestor) {
+            const duplicatedAncestors = this._splitUpToAncestor(last, lastListAncestor);
+            duplicatedNodes = [...duplicatedNodes, ...duplicatedAncestors];
+        }
+
+        // If the first node is within a list, we need to split that list.
+        // Then insert the new list to which the nodes will be appended.
+        const first = nodes[0];
+        const firstListAncestor = this._listAncestor(first);
+        if (firstListAncestor) {
+            const duplicatedAncestors = this._splitUpToAncestor(first, firstListAncestor);
+            duplicatedNodes = [...duplicatedNodes, ...duplicatedAncestors];
+
+            // Insert the new list before the first node's list ancestor.
+            // It was split so we need its duplicate as that is where the node is.
+            const newListAncestor = duplicatedAncestors[duplicatedAncestors.length - 1];
+            newListAncestor.before(newList);
+        } else {
+            // Insert the new list before the common ancestor to the first and
+            // last nodes.
+            const commonAncestor = this._commonAncestor(first, last);
+            const reference = first.ancestor(node => node.parent === commonAncestor) || first;
+            commonAncestor.insertBefore(newList, reference);
+        }
+
+        // Move the nodes to the list
+        withMarkers(() => {
+            // Remove the children of nested lists (they will be moved together
+            // with their parent) and convert nested lists to the new type.
+            const nestedLists = nodes.filter(this._isNestedList.bind(this));
```

Comment:
`_isNestedList` should not be on `this`, it should be a pure function, same for `listAncestor` then you will be able to remove the type castings.